### PR TITLE
Adapt layout for mobile (#48)

### PR DESCRIPTION
**Pull request by blms**
## In this PR

- Per #48:
  - Add toggle-able mobile navigation using a checkbox input and hamburger icon
  - Overall reductions in padding, margins for mobile
  - Adapt content, search pages for mobile
  - Adapt home page for mobile by using flex column and placing the `aside` content last
  - Adapt map controls for mobile
    - Use flex column to stack controls vertically
    - Use a smaller font size and flex row-wrap on author select
- For consistency (and accuracy!), replace all `React.Component` return types with `ReactElement` in JSDoc

### CODE REVIEW

File: src/components/AuthorSelect/AuthorSelect.css
Comment (rsimon) [inline src/components/AuthorSelect/AuthorSelect.css:8]: The only (nit-picky) thing I'm noticing is the way that `flex-flow: row wrap` will cause the labels to become mis-aligned. IMO a grid arrangement would look cleaner better. But with CSS grid, we can't define a grid without pre-setting the number of columns. Which means it's tricky on different devices. I.e.: it's tricky.

We could perhaps set a grid and switch between four and two columns, using different screen width breakpoints...

But in any case: that's just the icing on the cake.

File: src/components/Header/Header.css
Comment (rsimon) [inline src/components/Header/Header.css:116]: The responsive menu is great!

File: src/index.css
Comment (rsimon) [inline src/index.css:133]: I love that solution!

File: src/components/AuthorSelect/AuthorSelect.css
Comment (rsimon) [reply src/components/AuthorSelect/AuthorSelect.css:8]: P.S: alternatively, and if we want to save even more space on the map views, we could turn it into a drop-down menu (like the nav menu). But as I said: icing on the cake :-)

**Review by rsimon — APPROVED** [review]
That's excellent work!

File: src/components/AuthorSelect/AuthorSelect.css
Comment (blms) [reply src/components/AuthorSelect/AuthorSelect.css:8]: I agree with that, and was considering a grid as well! I think 3 columns might work on all non-desktop displays. I think I will merge this and once we get your PR #52 merged, I'll open a new one with revisions to the mobile display, since it is impacted by that PR.

**Pull request merged**
